Code/Framework/AzCore/AzCore/Settings/SettingsRegistryOriginTracker.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/std/optional.h>
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/std/containers/deque.h>
+#include <AzCore/std/containers/stack.h>
+#include <AzCore/std/function/function_template.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryImpl.h>
+#include <AzCore/DOM/DomPrefixTree.h>
+#include <AzCore/Serialization/Json/JsonSerializationResult.h>
+
+namespace AZ
+{
+    class SettingsRegistryOriginTracker
+    {
+    public:
+        AZ_TYPE_INFO(SettingsRegistryOriginTracker, "{F6E3B4E5-F8CF-43D8-8609-BC322D9D08C5}");
+
+        explicit SettingsRegistryOriginTracker(AZ::SettingsRegistryInterface& registry);
+        ~SettingsRegistryOriginTracker();
+
+        //! TrackingFilterCallback will be invoked with the JSON settings key being updated
+        //! It should return true if the key at that JSON path should be tracked
+        using TrackingFilterCallback = AZStd::function<bool(AZStd::string_view keyPath)>;
+
+        //! Sets functions which will be used to filter which settings should be tracked
+        //! If empty, all settings within the associated registry will be tracked
+        void SetTrackingFilter(TrackingFilterCallback filterCallback);
+
+        //! Gets setting registry being tracked
+        AZ::SettingsRegistryInterface& GetSettingsRegistry();
```

Comment:
Can you also add a version of this function works with a constant SettingsRegistryOriginTracker instance?

```suggestion
        //! Gets setting registry being tracked
        AZ::SettingsRegistryInterface& GetSettingsRegistry();
         const AZ::SettingsRegistryInterface& GetSettingsRegistry() const;
```